src/useLocalStore.ts

```diff
@@ -0,0 +1,22 @@
+import { observable } from "mobx"
+import { useState } from "react"
+
+import { useDeprecated } from "./utils"
+import { useAsObservableSource } from "./useAsObservableSource"
+
+export function useLocalStore<TStore extends Record<string, any>>(initializer: () => TStore): TStore
+export function useLocalStore<TStore extends Record<string, any>, TSource extends object>(
+    initializer: (source: TSource) => TStore,
+    current: TSource
+): TStore
+export function useLocalStore<TStore extends Record<string, any>, TSource extends object>(
+    initializer: (source?: TSource) => TStore,
+    current?: TSource
+): TStore {
+    if ("production" !== process.env.NODE_ENV)
+        useDeprecated(
+            "[mobx-react-lite] 'useLocalStore' is deprecated, use 'useObservable' instead"
```

Comment:
Wait, what? Do you really want to re-add `useObservable`? We had that in 0.x and it was super confusing with `useObserver` ... which now doesn't exist so I guess it might be ok, but these are huge shifts in paradigm.

Reply:
Mostly to be able to arg swap the `source` with `annotations`. But maybe we can do that without renaming, as TS and runtime validation will fail if people use it wrong

Reply:
Renamed to `useLocalObservable` and explained the difference in readme